### ISSUE TITLE
[6.0.0] '--rulesdir' option doesn't work if a config file has 'root:true'

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#11888]
**Tell us about your environment**

* **ESLint Version:** 6.0.0
* **Node Version:** 12.2.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Default.

**Please show your full configuration:**

```js
module.exports = {
    root: true,
    rules: {
        test: "error"
    }
};
```

```js
// internal-rules/test.js
const { Linter } = require("eslint")
const linter = new Linter()
const rules = linter.getRules()

module.exports = rules.get("no-console")
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.log("hello")
```

```bash
eslint test.js --rulesdir internal-rules
```

**What did you expect to happen?**

The `test` rule in the `internal-rules` directory warns the code.

**What actually happened? Please include the actual, raw output from ESLint.**

```
C:\Users\t-nagashima.AD\dev\sandbox\test.js
  1:1  error  Definition for rule 'test' was not found  test

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

Yes.
